### PR TITLE
TD-21220 fix assert

### PR DESCRIPTION
**Pull request by facetosea**

### CODE REVIEW

File: source/os/src/osFile.c [hunk 1]
```diff
@@ -561,19 +584,19 @@ int32_t taosUnLockFile(TdFilePtr pFile) {
...
-  assert(pFile->fd >= 0);  // Please check if you have closed the file.
-
   return (int32_t)flock(pFile->fd, LOCK_UN | LOCK_NB);
 #endif
 }
 
 int32_t taosFtruncateFile(TdFilePtr pFile, int64_t l_size) {
-#ifdef WINDOWS
-  if (pFile->fd < 0) {
-    errno = EBADF;
```
Comment (sangshuduo) [inline source/os/src/osFile.c:573]: no need to set errno as before?

File: source/os/src/osFile.c [hunk 2]
```diff
@@ -803,7 +828,10 @@ int32_t taosEOFFile(TdFilePtr pFile) {
   if (pFile == NULL) {
     return 0;
   }
-  assert(pFile->fp != NULL);
+  ASSERT(pFile->fp != NULL);
+  if(pFile->fp == NULL) {
+    return 0;
```
Comment (sangshuduo) [inline source/os/src/osFile.c]: Seems other function return -1 if pFile->fp is NULL

File: source/os/src/osMemory.c [hunk 3]
```diff
@@ -348,7 +356,7 @@ void taosMemoryTrim(int32_t size) {
 
 void* taosMemoryMallocAlign(uint32_t alignment, int64_t size) {
 #ifdef USE_TD_MEMORY
-  assert(0);
+  ASSERT(0);
```
Comment (sangshuduo) [inline source/os/src/osMemory.c:359]: It seems a useless legacy code. Should we just remove it?

File: source/os/src/osSocket.c [hunk 4]
```diff
@@ -298,7 +298,7 @@ int32_t taosGetSockOpt(TdSocketPtr pSocket, int32_t level, int32_t optname, void
     return -1;
   }
 #ifdef WINDOWS
-  assert(0);
+  ASSERT(0);
```
Comment (sangshuduo) [inline source/os/src/osSocket.c:301]: same here

File: source/os/src/osSysinfo.c [hunk 5]
```diff
@@ -863,7 +863,7 @@ int32_t taosGetSystemUUID(char *uid, int32_t uidlen) {
 
 char *taosGetCmdlineByPID(int pid) {
 #ifdef WINDOWS
-  assert(0);
+  ASSERT(0);
```
Comment (sangshuduo) [inline source/os/src/osSysinfo.c:866]: same here

File: tools/shell/src/shellAuto.c [hunk 6]
```diff
@@ -715,7 +715,10 @@ void putBackAutoPtr(int type, STire* tire) {
 
   } else {
     tires[type]->ref--;
-    assert(tires[type]->ref > 0);
+    ASSERT(tires[type]->ref > 0);
+    if (tires[type]->ref <= 0) {
+      return
```
Comment (sangshuduo) [inline tools/shell/src/shellAuto.c]: miss ;

File: tools/shell/src/shellCommand.c [hunk 7]
```diff
@@ -62,7 +62,8 @@ int32_t shellCountPrefixOnes(uint8_t c) {
 }
 
 void shellGetPrevCharSize(const char *str, int32_t pos, int32_t *size, int32_t *width) {
-  assert(pos > 0);
+  ASSERT(pos > 0);
+  if (pos <= 0) return;
```
Comment (sangshuduo) [inline tools/shell/src/shellCommand.c:66]: how about print a log for unexpected input here?

File: tools/shell/src/shellCommand.c [hunk 8]
```diff
@@ -75,21 +76,23 @@ void shellGetPrevCharSize(const char *str, int32_t pos, int32_t *size, int32_t *
...
-  // assert(rc == *size); // it will be core, if str is encode by utf8 and taos charset is gbk
+  // ASSERT(rc == *size); // it will be core, if str is encode by utf8 and taos charset is gbk
 
   *width = taosWcharWidth(wc);
 }
 
 void shellGetNextCharSize(const char *str, int32_t pos, int32_t *size, int32_t *width) {
-  assert(pos >= 0);
+  ASSERT(pos >= 0);
+  if(pos < 0) return;
```
Comment (sangshuduo) [inline tools/shell/src/shellCommand.c:86]: how about print a log for unexpected input here?

File: source/os/src/osSocket.c [hunk 9]
```diff
@@ -689,7 +689,7 @@ int taosGetLocalIp(const char *eth, char *ip) {
 int taosValidIp(uint32_t ip) {
 #if defined(WINDOWS)
   // DO NOTHAING
-  assert(0);
+  ASSERT(0);
```
Comment (sangshuduo) [inline source/os/src/osSocket.c:692]: same here

File: source/os/src/osSocket.c [hunk 10]
```diff
@@ -662,7 +662,7 @@ int32_t taosKeepTcpAlive(TdSocketPtr pSocket) {
 int taosGetLocalIp(const char *eth, char *ip) {
 #if defined(WINDOWS)
   // DO NOTHAING
-  assert(0);
+  ASSERT(0);
```
Comment (sangshuduo) [inline source/os/src/osSocket.c:665]: same here

File: source/os/src/osSocket.c [hunk 11]
```diff
@@ -994,7 +994,7 @@ int32_t taosGetFqdn(char *fqdn) {
 #else
     printf("failed to get hostname, reason:%s\n", strerror(errno));
 #endif
-    assert(0);
+    ASSERT(0);
```
Comment (sangshuduo) [inline source/os/src/osSocket.c:997]: same here

File: source/os/src/osString.c [hunk 12]
```diff
@@ -112,7 +112,7 @@ int32_t tasoUcs4Compare(TdUcs4 *f1_ucs4, TdUcs4 *f2_ucs4, int32_t bytes) {
 }
 
 TdUcs4 *tasoUcs4Copy(TdUcs4 *target_ucs4, TdUcs4 *source_ucs4, int32_t len_ucs4) {
-  assert(taosMemorySize(target_ucs4) >= len_ucs4 * sizeof(TdUcs4));
+  ASSERT(taosMemorySize(target_ucs4) >= len_ucs4 * sizeof(TdUcs4));
```
Comment (sangshuduo) [inline source/os/src/osString.c:115]: no need check?

File: source/os/src/osString.c [hunk 13]
```diff
@@ -355,8 +355,8 @@ int64_t taosStr2Int64(const char *str, char **pEnd, int32_t radix) {
   if (errno == EINVAL) errno = 0;
 #endif
 #ifdef TD_CHECK_STR_TO_INT_ERROR
-  assert(errno != ERANGE);
-  assert(errno != EINVAL);
+  ASSERT(errno != ERANGE);
+  ASSERT(errno != EINVAL);
```
Comment (sangshuduo) [inline source/os/src/osString.c:359]: how about print a log for unexpected input here?

File: source/os/src/osString.c [hunk 14]
```diff
@@ -367,8 +367,8 @@ uint64_t taosStr2UInt64(const char *str, char **pEnd, int32_t radix) {
...
```
Comment (sangshuduo) [inline source/os/src/osString.c:371]: how about print a log for unexpected input here?

File: source/os/src/osString.c [hunk 15]
```diff
@@ -379,8 +379,8 @@ int32_t taosStr2Int32(const char *str, char **pEnd, int32_t radix) {
...
```
Comment (sangshuduo) [inline source/os/src/osString.c:383]: how about print a log for unexpected input here?

File: source/os/src/osString.c [hunk 16]
```diff
@@ -391,8 +391,8 @@ uint32_t taosStr2UInt32(const char *str, char **pEnd, int32_t radix) {
...
```
Comment (sangshuduo) [inline source/os/src/osString.c:395]: how about print a log for unexpected input here?

File: source/os/src/osString.c [hunk 17]
```diff
@@ -403,10 +403,10 @@ int16_t taosStr2Int16(const char *str, char **pEnd, int32_t radix) {
...
 #endif
 #ifdef TD_CHECK_STR_TO_INT_ERROR
-  assert(errno != ERANGE);
-  assert(errno != EINVAL);
-  assert(tmp >= SHRT_MIN);
-  assert(tmp <= SHRT_MAX);
+  ASSERT(errno != ERANGE);
+  ASSERT(errno != EINVAL);
+  ASSERT(tmp >= SHRT_MIN);
+  ASSERT(tmp <= SHRT_MAX);
```
Comment (sangshuduo) [inline source/os/src/osString.c:409]: how about print a log for unexpected input here?

File: source/os/src/osString.c [hunk 18]
```diff
@@ -417,20 +417,20 @@ uint16_t taosStr2UInt16(const char *str, char **pEnd, int32_t radix) {
   if (errno == EINVAL) errno = 0;
 #endif
 #ifdef TD_CHECK_STR_TO_INT_ERROR
-  assert(errno != ERANGE);
-  assert(errno != EINVAL);
-  assert(tmp <= USHRT_MAX);
+  ASSERT(errno != ERANGE);
+  ASSERT(errno != EINVAL);
+  ASSERT(tmp <= USHRT_MAX);
```
Comment (sangshuduo) [inline source/os/src/osString.c:422]: how about print a log for unexpected input here?

File: source/os/src/osString.c [hunk 19]
```diff
@@ -417,20 +417,20 @@ uint16_t taosStr2UInt16(const char *str, char **pEnd, int32_t radix) {
...
   int32_t tmp = strtol(str, pEnd, radix);
 #ifdef TD_CHECK_STR_TO_INT_ERROR
-  assert(errno != ERANGE);
-  assert(errno != EINVAL);
-  assert(tmp >= SCHAR_MIN);
-  assert(tmp <= SCHAR_MAX);
+  ASSERT(errno != ERANGE);
+  ASSERT(errno != EINVAL);
+  ASSERT(tmp >= SCHAR_MIN);
+  ASSERT(tmp <= SCHAR_MAX);
```
Comment (sangshuduo) [inline source/os/src/osString.c:433]: how about print a log for unexpected input here?

File: source/os/src/osString.c [hunk 20]
```diff
@@ -441,30 +441,30 @@ uint8_t taosStr2UInt8(const char *str, char **pEnd, int32_t radix) {
   if (errno == EINVAL) errno = 0;
 #endif
 #ifdef TD_CHECK_STR_TO_INT_ERROR
-  assert(errno != ERANGE);
-  assert(errno != EINVAL);
-  assert(tmp <= UCHAR_MAX);
+  ASSERT(errno != ERANGE);
+  ASSERT(errno != EINVAL);
+  ASSERT(tmp <= UCHAR_MAX);
```
Comment (sangshuduo) [inline source/os/src/osString.c:446]: how about print a log for unexpected input here?

File: source/os/src/osString.c [hunk 21]
```diff
@@ -441,30 +441,30 @@ uint8_t taosStr2UInt8(const char *str, char **pEnd, int32_t radix) {
...
 
 double taosStr2Double(const char *str, char **pEnd) {
   double tmp = strtod(str, pEnd);
 #ifdef TD_CHECK_STR_TO_INT_ERROR
-  assert(errno != ERANGE);
-  assert(errno != EINVAL);
-  assert(tmp != HUGE_VAL);
+  ASSERT(errno != ERANGE);
+  ASSERT(errno != EINVAL);
+  ASSERT(tmp != HUGE_VAL);
```
Comment (sangshuduo) [inline source/os/src/osString.c:456]: how about print a log for unexpected input here?

File: source/os/src/osString.c [hunk 22]
```diff
@@ -441,30 +441,30 @@ uint8_t taosStr2UInt8(const char *str, char **pEnd, int32_t radix) {
...
   float tmp = strtof(str, pEnd);
 #ifdef TD_CHECK_STR_TO_INT_ERROR
-  assert(errno != ERANGE);
-  assert(errno != EINVAL);
-  assert(tmp != HUGE_VALF);
-  assert(tmp != NAN);
+  ASSERT(errno != ERANGE);
+  ASSERT(errno != EINVAL);
+  ASSERT(tmp != HUGE_VALF);
+  ASSERT(tmp != NAN);
```
Comment (sangshuduo) [inline source/os/src/osString.c:467]: how about print a log for unexpected input here?

File: tools/shell/src/shellEngine.c [hunk 23]
```diff
@@ -713,7 +713,7 @@ int32_t shellCalcColWidth(TAOS_FIELD *field, int32_t precision) {
       }
 
     default:
-      assert(false);
+      ASSERT(false);
```
Comment (sangshuduo) [inline tools/shell/src/shellEngine.c:716]: useless code

**Review by sangshuduo — CHANGES_REQUESTED** [review]
see my comments

File: source/os/src/osFile.c (hunk 2 above)
Comment (facetosea) [reply source/os/src/osFile.c]: 这个和上面pFile==NULL的行为一致；不过看了下，两个都返回-1更合适，类似feof本身的行为. done

**Review by sangshuduo — APPROVED** [review]

**Pull request merged**